src/frontends/pytorch/src/op/index_put_.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/scatter_nd_update.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/slice.hpp"
+#include "openvino/op/split.hpp"
+#include "openvino/op/unsqueeze.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+namespace {
+Output<Node> generate_zeros_with_convertlike(NodeContext& context, Output<Node> sizes, Output<Node> tensor_of_type) {
+    auto const_0 = context.mark_node(v0::Constant::create(element::i32, Shape{}, {0}));
+    auto zeros = context.mark_node(std::make_shared<v3::Broadcast>(const_0, sizes));
+    return context.mark_node(std::make_shared<v1::ConvertLike>(zeros, tensor_of_type));
+}
+}  // namespace
+
+OutputVector translate_index_put_(NodeContext& context) {
```

Comment:
Please check number of inputs using `num_inputs_check` helper

Reply:
done